src/test/scala/tech/cryptonomic/conseil/metadata/MetadataServiceTest.scala

```diff
@@ -4,39 +4,50 @@ import akka.http.scaladsl.testkit.ScalatestRouteTest
 import org.scalamock.scalatest.MockFactory
 import org.scalatest.concurrent.ScalaFutures
 import org.scalatest.time.{Millis, Seconds, Span}
-import org.scalatest.{Matchers, WordSpec}
+import org.scalatest.{BeforeAndAfterEach, Matchers, WordSpec}
 import tech.cryptonomic.conseil.config.Platforms.{PlatformsConfiguration, TezosConfiguration, TezosNodeConfiguration}
 import tech.cryptonomic.conseil.config.Types.PlatformName
 import tech.cryptonomic.conseil.config._
 import tech.cryptonomic.conseil.generic.chain.PlatformDiscoveryTypes.DataType.{Hash, Int}
 import tech.cryptonomic.conseil.generic.chain.PlatformDiscoveryTypes.KeyType.NonKey
 import tech.cryptonomic.conseil.generic.chain.PlatformDiscoveryTypes.{Attribute, Entity, Network, Platform}
-import tech.cryptonomic.conseil.tezos.TezosPlatformDiscoveryOperations
 
-import scala.concurrent.Future.successful
+import scala.concurrent.Future
 
-class MetadataServiceTest extends WordSpec with Matchers with ScalatestRouteTest with MockFactory with ScalaFutures {
+class MetadataServiceTest
+    extends WordSpec
+    with Matchers
+    with ScalatestRouteTest
+    with MockFactory
+    with ScalaFutures
+    with BeforeAndAfterEach {
 
   implicit override val patienceConfig = PatienceConfig(timeout = Span(2, Seconds), interval = Span(20, Millis))
 
-  "The metadata service" should {
-
-      val tezosPlatformDiscoveryOperations = stub[TezosPlatformDiscoveryOperations]
-      val cacheOverrides = stub[AttributeValuesCacheConfiguration]
+  // shared objects
+  var platformDiscoveryOperations = new TestPlatformDiscoveryOperations
+  val cacheOverrides = stub[AttributeValuesCacheConfiguration]
 
-      val sut = (metadataOverridesConfiguration: Map[PlatformName, PlatformConfiguration]) =>
-        new MetadataService(
-          PlatformsConfiguration(
-            Map(
-              Platforms.Tezos -> List(
-                    TezosConfiguration("mainnet", TezosNodeConfiguration("tezos-host", 123, "https://"))
-                  )
-            )
-          ),
-          new UnitTransformation(MetadataConfiguration(metadataOverridesConfiguration)),
-          cacheOverrides,
-          tezosPlatformDiscoveryOperations
+  val sut = (metadataOverridesConfiguration: Map[PlatformName, PlatformConfiguration]) =>
+    new MetadataService(
+      PlatformsConfiguration(
+        Map(
+          Platforms.Tezos -> List(
+                TezosConfiguration("mainnet", TezosNodeConfiguration("tezos-host", 123, "https://"))
+              )
         )
+      ),
+      new UnitTransformation(MetadataConfiguration(metadataOverridesConfiguration)),
+      cacheOverrides,
+      platformDiscoveryOperations
+    )
+
+  override protected def beforeEach(): Unit = {
+    super.beforeEach()
+    platformDiscoveryOperations.clear()
```

Comment:
instead of that you could just mix in `OneInstancePerTest`, but I guess it depends on preferences

Reply:
done